src/branch_and_prune.jl

```diff
@@ -221,9 +181,45 @@ Inputs:
 - `X`: `Interval` or `IntervalBox`
 - `contractor`: function that, when applied to the function `f`, determines
     the status of a given box `X`. It returns the new box and a symbol indicating
-    the status. Current possible values are `Bisection` and `Newton`.
+    the status. Current possible values are `Bisection`, `Newton` and
+    `Newton(f_prime)` where `f_prime` is the derivative or jacobian of `f`.
 
 """
-roots{C<:Contractor}(f, X, contractor::Type{C}, tol::Float64=1e-3) = branch_and_prune(f, X, contractor, tol)
+function roots{T, C<:Contractor}(f, X::IntervalLike{T}, contractor::Type{C}, tol::Float64=1e-3)
+    dim = length(X)
```

Comment:
Can the length of `X` just be a type parameter of the `roots` function?

```
+function roots(f, X::IntervalBox{T,N}, contractor::Type{C}, tol::Float64=1e-3) where {T, N, C<:Contractor}
```
and just wrap an Interval in a 1D IntervalBox? Or just provide a separate function altogether for the 1D case.

Reply:
(Please use the future-proof `where` syntax.)